sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+      }
+      else
+      {
+        send_reported_temperature_property(desired_temp, version_num);
+
+        if (update_device_temp((int32_t)desired_temp))
+        {
+          send_reported_temperature_property(device_max_temp, version_num);
+        }
+      }
+      if (message->payloadlen)
+      {
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+      }
+      break;
+    // An update to the desired properties with the properties as a JSON payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+      printf("A twin desired properties message was received\n");
+      printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+
+      // Get the new temperature
+      if (az_failed(
+              result = parse_twin_desired_temperature_property(
+                  twin_payload_span, false, &desired_temp, &version_num)))
+      {
+        printf("Could not parse desired temperature property, az_result %04x\n", result);
+        break;
+      }
+      send_reported_temperature_property(desired_temp, version_num);
+
+      if (update_device_temp((int32_t)desired_temp))
+      {
+        send_reported_temperature_property(device_max_temp, -1);
+      }
+      break;
+
+    // A response from a twin reported properties publish message. With a successfull update of
+    // the reported properties, the payload will be empty and the status will be 204.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      printf("A twin reported properties response message was received\n");
+      break;
+  }
+}
+
+// Invoke the requested command if supported and return status | Return error otherwise
+static void handle_command_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_method_request* command_request)
+{
+
+  if (az_span_is_content_equal(report_command_name_span, command_request->name))
+  {
+    az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+    az_span command_payload_span
+        = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+    // Invoke command
+    uint16_t return_code;
+    az_result response = invoke_getMaxMinReport(
+        command_payload_span, command_response_span, &command_response_span);
+    if (response == AZ_ERROR_ITEM_NOT_FOUND)
+    {
+      return_code = 404;
+    }
+    else
+    {
+      return_code = 200;
+    }
+
+    // Send command response with report as JSON payload
+    int rc;
+    if ((rc = send_command_response(command_request, return_code, command_response_span)) != 0)
+    {
+      printf("Unable to send %u response, status %d\n", return_code, rc);
+    }
+  }
+  else
+  {
+    // Unsupported command
+    printf(
+        "Unsupported command received: %.*s.\n",
+        az_span_size(command_request->name),
+        az_span_ptr(command_request->name));
+
+    int rc;
+    if ((rc = send_command_response(command_request, 501, report_error_payload)) != 0)
```

Comment:
>501 [](start = 53, length = 3)

We should sync at some point with all sdks for error code alignment, but this feels like a 404.  From wikipedia [https://en.wikipedia.org/wiki/List_of_HTTP_status_codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes), a 501 => "Usually this implies future availability ".  This one is the requestor's fault (4xx) not ours :)

Reply:
True. I saw Himanshu had it implemented as a `501` and that at the time seemed more fitting. The
> Usually this implies future availability

makes me think twice. I can revert for now and we can sync on that.